js/block-editor.js

```diff
@@ -50,56 +56,77 @@ function getCurrentLanguage() {
  */
 jQuery(
 	function( $ ) {
+		// Initialize current language to be able to compare if it changes.
+		initializeLanguageOldValue();
+
+
 		// Ajax for changing the post's language in the languages metabox
 		$( '.post_lang_choice' ).on(
 			'change',
 			function( event ) {
 				const select = wp.data.select;
 				const dispatch = wp.data.dispatch;
 				const subscribe = wp.data.subscribe;
-				const title = select( 'core/editor' ).getEditedPostAttribute( 'title' );
-				const content = select( 'core/editor' ).getEditedPostAttribute( 'content' );
-				const excerpt = select( 'core/editor' ).getEditedPostAttribute( 'excerpt' );
+				const emptyPost = bypassConfirmation();
+
+				// Initialize the confirmation dialog box.
+				const { dialogContainer : dialog, dialogResult } = initializeConfimationModal();
+				// The selected option in the dropdown list.
+				const selectedOption = event.target;
 
 				// Specific case for empty posts.
 				// Place at the beginning because window.location changing triggers automatically page reloading.
-				if ( location.pathname.match( /post-new.php/gi ) && '' === title && '' === content && '' === excerpt ) {
-					reloadPageForEmptyPost( this.value );
+				if ( location.pathname.match( /post-new.php/gi ) && emptyPost ) {
+					reloadPageForEmptyPost( selectedOption.value );
 				}
 
 				// Otherwise send an ajax request to refresh the legacy metabox and set the post language with the new language.
+				// It needs a confirmation of the user before changing the language.
 				// Need to wait the ajax response before triggering the block editor post save action.
-				var data = {
-					action:     'post_lang_choice',
-					lang:       event.target.value,
-					post_type:  $( '#post_type' ).val(),
-					post_id:    $( '#post_ID' ).val(),
-					_pll_nonce: $( '#_pll_nonce' ).val()
+				if ( $( this ).data( 'old-value' ) !== selectedOption.value && ! emptyPost ) {
+					dialog.dialog( 'open' );
+				} else {
+					// Block editor doesn't allow to save an empty post.
+					// So we must revert the language to the old one because the browser asks if we want to quit the page if we trigger its reloading.
+					selectedOption.value = $( this ).data( 'old-value' );
```

Comment:
I modified to accept the language changing in case 2 - `Promise.resolve()`. But I think we need to reload the page especially because the block editor don't save an empty post. 

However it's a limit case when the legacy metabox is used in Polylang Pro by using the PLL_USE_BLOCK_EDITOR_PLUGIN constant and to refresh correctly categories, tags or parent page. It's impossible now with Polylang which doesn't implement REST API to filter categories, tags and parent page by language.

This reload can't be reload as a new post because we don't need to inject the `new_lang` paremeter.

In the other hand if we reload the page in this case and accept the language changing, it's because the user emptied his post. The browser warns about that and asks if he want to reload the page. So we can't do the reloading programmatically without a user confirmation.

So we can consider acceptable that just change the language without reloading the page is enough.